.github/workflows/test.yml

```diff
@@ -19,28 +19,40 @@ jobs:
       fail-fast: false
       matrix:
         env: ${{ fromJSON(needs.getcidata.outputs.environments) }}
-        include:
-          - services: {}
-            redis_host: localhost
-          - env: { os: ubuntu-latest }
-            services: { redis: { image: redis, ports: [ '6379:6379' ] } }
-            redis_host: redis
     runs-on: ${{ matrix.env.os }}
     container: ${{ matrix.env.image }}
-    services: ${{ matrix.services }}
     steps: 
       - name: Select toolchain
         uses: maxim-lobanov/setup-xcode@v1.2.1
         with:
           xcode-version: ${{ matrix.env.toolchain }}
         if: ${{ matrix.env.toolchain != '' }}
-      - name: Set up Redis server via Homebrew
-        if: ${{ matrix.env.toolchain != '' }}
-        run: brew install redis && brew services start redis
-      - name: Check out Redis
+      - name: Check out source code
         uses: actions/checkout@v2
+
+      - name: Install Redis via Homebrew (Mac)
+        if: ${{ matrix.env.toolchain != '' }}
+        run: brew install redis 
+
+      - name: Install Redis via apt-get (Ubuntu)
+        if: ${{ matrix.env.toolchain == '' }}
+        run: apt-get update && apt-get install -y redis
+
+      - name: Start Redis 1
+        run: mkdir -p /tmp/redis1 && redis-server .github/redis1.conf
+      - name: Start Redis 2
+        run: mkdir -p /tmp/redis2 && redis-server .github/redis2.conf
```

Comment:
Are these configurations necessary instead of the defaults?

If we were to use the Docker containers then we can assign the values as environment variables

Reply:
there's a few problems, unfortunately.
1.  Docker isn't supported in GitHub Actions for Mac. 
2. Brew doesn't support running multiple redis instances

So since we're already making separate redis.conf files for Mac, it made sense to use them on Ubuntu, as well. So the build scripts are consistent.

Reply:
I'd be OK with just testing this on Linux w/ Docker & GitHub Action Services since it's a single use case

@gwynne Might have some ideas or a different opinion though

Reply:
Since Mac is a supported platform, I think it is good for the tests to run on Mac as well.

Reply:
@Mordil @gwynne  I put some effort into simplifying the redis conf files. Nobody wants massive config files 😂 . It's down to 4 lines: 1. run in background, 2. pidfile, 3. port, 4. turn off logging.

Hopefully, this effort helps us keep both Mac and Linux as fully tested platforms for this package. 

(Of course I would prefer to use Docker for both platforms, but unfortunately, Docker is not supported for Mac on GitHub Actions)